### ISSUE TITLE
Deletes only every other item from a list

### ISSUE THREAD

**Issue by Diaoul** [mikefarah/yq#702]
**Describe the bug**
Deletion occurs only on every other item of a list

Version of yq: 4.4.1
Operating system: Linux
Installed via: AUR

**Input Yaml**
test.yaml:
```yaml
list:
  - elem: "1"
    sublist: ["a", "b"]
  - elem: "2"
    sublist: ["c", "d"]
  - elem: "3"
    sublist: ["e", "f"]
  - elem: "4"
    sublist: ["g", "h"]
```

**Command**
```
yq eval 'del(.list.[] | select(true))' test.yaml
```

**Actual behavior**

```yaml
list:
  - elem: "2"
    sublist: ["c", "d"]
  - elem: "4"
    sublist: ["g", "h"]
```

**Expected behavior**

```yaml
list: []
```

**Comment by mikefarah** [comment]
Yeah looks like a bug :(

for the moment, you can run:

```
yq eval '.list = []'
```

for the record, you don't need to pipe through to select(true) - this has no effect

**Comment by mikefarah** [comment]
Fixed in 4.5.0!
